### PR TITLE
Remove map state

### PR DESCRIPTION
**Pull request by lfender6445**
Destroy the entire map on unmount

### CODE REVIEW

File: packages/react-ui-map/src/Mapbox/Mapbox.js [hunk 1]
```diff
@@ -67,7 +67,11 @@ export default class Mapbox extends PureComponent {
 
   shouldComponentUpdate(nextProps, nextState) {
     return (nextProps.children !== this.props.children ||
-      nextState.map !== this.state.map)
+      nextState.map !== this.map)
```
Comment (tcwilson) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: this condition, `nextState.map !== this.map` no longer makes sense. As `map` isn't part of state, and `this.map` should get destroyed on component unmount anyway

**Comment by paulpostel** [comment]
I'm not really sure what the issue was that led to this.  This seems to mainly delay initialization to `ComponentDidMount`  the PR description says `Destroy the entire map on unmount` is that happening?

**Comment by lfender6445** [comment]
the PR description says Destroy the entire map on unmount is that happening?

yes at `componentWillUnmount -> map.remove()`

There was actually 2 issues - 

  - map was not loading reliably or consistently (hence the map on styleload change)
  - map resources were not being cleaned up properly between unmounts, so rather than worry about any of them, we destroy the entire map instead.

File: packages/react-ui-map/src/Mapbox/Mapbox.js [hunk 2]
```diff
@@ -111,7 +113,7 @@ export default class Mapbox extends PureComponent {
             theme.Map,
           )}
         />
-        {map && children}
+        {(this.state.loaded && this.map) && children}
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: You can probably remove the parens.

**Comment by paulpostel** [comment]
👍

**Pull request merged**